<?xml version="1.0"?>
<launch>

   <arg name="camera_aravis_package" default="true"/>
   <arg name="camera_number" default="1"/>
   <arg name="basler_sn" default="21511088"/>
   <arg name="camera_name" default="Basler1"/>
   <arg name="camera_id" default="Basler-$(arg basler_sn)"/>
   <arg name="guid" default="Basler-$(arg basler_sn)" />
   <arg name="exposure_auto" default="Once"/>
   <arg name="exposure_mode" default="auto_once"/>
   <arg name="exposure_time" default="2000.0" />
   <arg name="auto_white_balance" default="true"/>
   <arg name="gain_auto" default="Once" />
   <arg name="gain" default="0" />
   <arg name="gain_mode" default="auto_once"/>
   <arg name="frame_rate" default="20.0" />
   <arg name="frame_id" default="$(arg camera_name)_optical_frame"/>
   <arg name="trigger_mode" default="Off" />
   <arg name="trigger_source" default="Line1" />
   <arg name="pixel_format" default="Mono8"/>
   <arg name="digital_shift" default="0"/>
   <arg name="mtu" default="1500"/>
   <arg name="bin_x" default="1"/>
   <arg name="bin_y" default="1"/>
   <arg name="roi_x" default="0"/>
   <arg name="roi_y" default="0"/>
   <arg name="roi_width" default="1600"/>
   <arg name="roi_height" default="1200"/>
   <arg name="url" default="file://$(find ind_cal_multi_camera)/camera_info/Basler$(arg basler_sn).yaml"/>


  <!--include  file="$(find ind_cal_multi_camera)/launch/basler_camera.launch" >
      <arg name="camera_name" value="Basler1"/>
      <arg name="camera_aravis_package" value="true"/>
      <arg name="basler_sn" value="$(arg basler_sn)"/>
      <arg name="exposure_auto" value="auto_once"/>
      <arg name="exposure_time" value="2000"/>
      <arg name="gain_auto" value="Once" />
      <arg name="gain" value="0" />
      <arg name="frame_rate" value="20"/>
      <arg name="trigger_mode" value="Off"/>
      <arg name="trigger_source" value="Line1"/>
      <arg name="pixel_format" value="Mono8"/>
  </include-->

     <node pkg="intrinsic_cal" type="rail_ical" name="rail_intrinsic_calibration" output="screen" >
          <!--param name="image_topic" value="/o3d3xx/camera/amplitude" /-->
          <param name="image_topic" value="/camera/image_raw" />
          <param name="camera_name" value="camera" />
          <param name="target_rows" value="20" />
          <param name="target_cols" value="33" />
          <param name="target_circle_dia" value="0.0135" />
          <param name="target_spacing" value="0.0254" />
          <param name="num_camera_locations" value="12" />
          <param name="camera_spacing" value=".025" />
          <!--param name="image_height" value="132" />
          <param name="image_width" value="176" /-->
          <param name="image_height" value="1200" />
          <param name="image_width" value="1600" />

          <param name="target_to_rail_distance" value="0.490" />
          <param name="use_circle_detector" value="true"/>
          <remap from="color_image" to="/camera/image_raw" />
     </node>

     <node pkg="nodelet" type="nodelet" name="image_proc_nodelet$(arg camera_number)"  args="manager"/>
     <node pkg="nodelet" type="nodelet" name="rectify$(arg camera_number)" args="load image_proc/rectify image_proc_nodelet$(arg camera_number)">
        <remap from="camera_info" to="Basler$(arg camera_number)/camera_info"/>
        <remap from="image_mono" to="Basler$(arg camera_number)/image_raw"/>
        <remap from="image_rect" to="Basler$(arg camera_number)/image_rect"/>
     </node>

  <node name="B1_view" pkg="image_view" type="image_view" >
      <remap to="/camera/image_raw" from="image"/>
  </node>

  <node name="observation_view" pkg="image_view" type="image_view" >
      <remap to="/observer_results_image" from="image"/>
  </node>

</launch>
